### ISSUE TITLE
Specifying a date when you're likely to request an item

### ISSUE THREAD

**Issue by cbowskill** [wellcomecollection/wellcomecollection.org#7266]
# What is it and who's it for?
The ability to allow users to select a likely date they'll come into the building and view the item when they're requesting an item (will need to be toggled independently of main requesting functionality)

# User story

**As a signed-in library member
I want to be able to choose a date to see an item of interest
So that I can plan my journey to the library accordingly**

Scenario: Library member selects a future date to visit the library

Given I'm a signed-in library member
When I'm on a works page
and I click to request an item
When I select the calendar area to choose a specific date
Then the system shows what dates are available to choose from
And displays an error message if I choose an unavailable date
Then the system submits my chosen date 
And the system shows the 'request confirmed' message

**As a signed-in library member
I want to be able to review what date(s) I booked to see requested items
So that I can book a library ticket for the correct day(s)**

Scenario: Signed-in user checks their item requested list

Given I'm a signed-in library member
When I view my library account page
Then the system shows the Item Requests list
And the system shows the 'pickup date requested' in the Item Requests section

**As a member of library staff
I want to know what specific date(s) a member plans to attend the RMR 
so that I can prioritise what items need to be produced each day.**

**As a Product Manager
I want the ability to turn on/off the date to collect functionality
so that we can choose to turn it on when current ticketing system is removed**

# Implementation 

## Acceptance criteria
1. Users are able to select an available date (based on business rules on what dates for collection should be available)
2. Users are able to see what dates are available / unavailable to select
3. If a user selects an unavailable date they're advised why it's unavailable and prompted to re-select
4. A user is able to see what date they've selected
5. Library staff are able to see what date a user is planning on coming into the library
6. We're able to toggle this feature on/off independently of request functionality
7. We're able to measure interactions with this feature

## Questions
<---Delete as appropriate--->
- What are all the current business rules that that determine dates available
- Is this mandatory or will we provide an option for not selecting any date at this point?

**Comment by cbowskill** [comment]
Here's the related comment about potential solution for doing the Sierra element: https://github.com/wellcomecollection/wellcomecollection.org/issues/6537#issuecomment-897614171

**Comment by davidpmccormick** [comment]
What should the default date be? My guess is the next available date, i.e. the next day that isn't a Sunday?

Although since Item requests need to be made by 10am the day before your visit, should we disable the next day if it has already gone 10am?

**Comment by cbowskill** [comment]
Yes, at a bare minimum rules to consider need to be:

- Library is open [how do we deal with Christmas closures, bank holidays etc. to make sure these don't show as available days]
- If it's pre 10am next day/24hrs if past 10am it would need to be the day after/48hrs
- A 2 week window from the time of the request (typical library process is they'd clear the holds/shelves after 2 weeks and they're likely to return to this in January)

A question about the journey for @DominiqueMarshall:
-Do we want to prevent people from requesting if they don't select a date or do we need an alternative/'i'm not sure' route. How do we deal with those

Other considerations (that i'm hoping will just naturally not be an issue but i've seen be an issue in other places...):
- That it can handle other timezones/people externally to the UK
- That it can handle daylight savings times

Is it worth Dom, yourself & I catching up on this and breaking this word done into clearer user stories?

**Comment by davidpmccormick** [comment]
I think we already have dates that the venues will be closed stored/updatable in Prismic somewhere – will check, but hopefully we'll be able to thread those dates into the logic here.

I'm also (unrealistically?) hoping the timezones/daylight saving thing won't be an issue since the dates and the times are always relative to what the time is in the library, not on the user's machine. But I think we will still need to account for that.

**Comment by DominiqueMarshall** [comment]
I'm not sure if we would prevent them from requesting, but I'll double-check that expectation with David L. I imagine if they don't select a date, a member of the library team will end up contacting them to get an idea of when they expect to visit? Need to confirm that.

**Comment by cbowskill** [comment]
Timezone/Daylight saving is just a scenario we try and test I think, to ensure it does work as expected.

@DominiqueMarshall - Could you work on a visual/flow that provides an alternative to selecting a date? I don't feel we should gate requesting behind selecting a date, particularly as some people may not know until later a specific day. How were they managing this before ticketing (as i don't believe you selected a date on the old site)?

**Comment by DominiqueMarshall** [comment]
There wasn't any date management before ticketing, or on the old site. Previously if you requested something, it would be picked by a member of staff within 24/48 hours, and it would be in the library for 2 weeks. If you didn't make it in to see it within that time frame, it would be taken back to storage and you would have to request it again. This is the first time any functionality like this will be created as a part of the process

**Comment by davidpmccormick** [comment]
We could (should?) have the next available date pre-selected in any event, so if you choose not to select anything, that is what will be sent.

**Comment by cbowskill** [comment]
I guess that would prompt the pick - did you say you were talking to David L @DominiqueMarshall . Might be worth checking he'd be happy with that as an approach. Feels logical to me.

**Comment by DominiqueMarshall** [comment]
@cbowskill So from David and Cassandra's perspective, we WOULD actually prevent people from requesting if they didn't choose a date. They would want to avoid any unsustainable workloads or admin burdens on the team having to chase members about it. There's also the possibility of users being disappointed if their items hadn’t been retrieved because they hadn’t selected a date.

I guess the flip side of the next available date being pre-selected is if no one changes that date and LE&E end up retrieving lots of items that aren't actually going to be picked up. That could be a non-issue, but definitely something worth testing.

Would any major issues arise from a technical perspective if we were to block people from requesting?

**Comment by davidpmccormick** [comment]
> Would any major issues arise from a technical perspective if we were to block people from requesting?

Don't think so. We could make the date input a required field, disallowing form submission without it.

Then the question is, should we pre-fill the date input with the next available date, or force the user to enter it manually?

If we pre-fill it, I guess there's a chance that someone will submit a request without realising there is a time-sensitive aspect to the collection (which might mean more work for LE&E).

If we don't pre-fill it, it's another click for the user, but more deliberate maybe?

If we don't pre-fill it, I think there's also the question of what the placeholder text in the input should be – if it already looks like an actual date, then you might think that it's pre-filled when it isn't. So something more like 'select a date', maybe?

**Comment by DominiqueMarshall** [comment]
Yeah, we'd want people to be deliberate about picking a date if it ends up being a required field. "Select a date" would be good as the alternative placeholder 👍🏽

**Comment by gestchild** [comment]
Closing in favour of #7756, which is the final piece of functionality required
